### ISSUE TITLE
Dynamic rendering mode

### ISSUE THREAD

**Issue by est31** [djc/askama#491]
Inspired by concerns voiced in [this comment](https://github.com/rust-lang/rust/issues/84419#issuecomment-826415318) that askama has a bad edit cycle because it involves recompilation each time. Ideally, to solve this, one could have a second rendering engine that runs during runtime. See also @djc 's [response](https://github.com/rust-lang/rust/issues/84419#issuecomment-857618378) in the thread.

**Comment by djc** [comment]
See also #389, #273.

**Comment by SimonAdameit** [comment]
This is something that I'd be interested in for another use case: I'd like to use the same template engine for static and dynamic parts of a web site. While the dynamic parts are fine to be statically compiled into a new binary, static parts should just be able to be generated by running a cmd tool and not involve compilation.

**Comment by djc** [comment]
See also #425 for more discussion of potential designs and a bunch of workarounds. We should really probably coalesce all of these issues into a single one. Anyway, something that's not changed is that I'm mostly unlikely to start work on solving this. That said, now that askama_parser is available as a standalone crate (on main only for now) this might make things like this easier.

**Comment by kellpossible** [comment]
One use case which I haven't seen mentioned here yet is the ability to override templates at runtime to allow user customizable templates

**Comment by djc** [comment]
Folks interested in this issue might like @rdbo's [dynja](https://github.com/rdbo/dynja), which uses minijinja for the debug profile and Askama for the release profile.

**Comment by zys864** [comment]
Is there a way to compile the template at runtime?e.g. compile a template from sql,then use it to render html?

**Comment by djc** [comment]
No, Askama's code generator generates Rust code and I don't think there are any good ways of compiling and then executing Rust code at runtime. You might have better luck with something like Tera (although if you just want to make templates depend on your SQL schema that sounds like it could be solved with code generation).
